src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private Storage storage;
+
+    public ReportGeneratorImpl(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public List<String> generate() {
+        Map<String,Integer> fruits = storage.getFruits();

```

Comment:
Right now we do not have much of the differance between using a direct Storage(DAO) and a FruitService, but there will be later on, and it is better to use service methods instead, also to be consistent with the handlers, that are using service as well, if there is no `getFruits()` method - let's just add